pkgs/applications/networking/mailreaders/himalaya/default.nix

```diff
@@ -0,0 +1,40 @@
+{ lib, stdenv, rustPlatform, fetchFromGitHub, openssl, pkgconfig, installShellFiles }:
+rustPlatform.buildRustPackage rec {
+  pname = "himalaya";
+  version = "0.2.6";
+
+  src = fetchFromGitHub {
+    owner = "soywod";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "1fl3lingb4wdh6bz4calzbibixg44wnnwi1qh0js1ijp8b6ll560";
+  };
+
+  cargoSha256 = "10p8di71w7hn36b1994wgk33fnj641lsp80zmccinlg5fiwyzncx";
+
+  nativeBuildInputs = [ openssl pkgconfig installShellFiles ];
+  buildInputs = [ openssl ];
+
+  postInstall = ''
+    # Install shell function
+    mkdir -p $out/completions
+
+    $out/bin/himalaya completion bash > $out/completions/himalaya.bash
+    installShellCompletion --bash $out/completions/himalaya.bash
+
+    $out/bin/himalaya completion fish > $out/completions/himalaya.fish
+    installShellCompletion --fish $out/completions/himalaya.fish
+
+    $out/bin/himalaya completion zsh > $out/completions/_himalaya
+    installShellCompletion --zsh $out/completions/_himalaya
+
+    rm -r $out/completions
+  '';
```

Comment:
Is there a better way of getting these completions? I doubt calling the resulting binary on the building machine will work well with cross-compilation.

Reply:
Doesn't look like it, some library they use dynamically generates them.

It definitely breaks cross-compilation though.

```
bt1cn@pcclu01:~/devel/nixpkgs$ nix-build . -A pkgsCross.aarch64-multiplatform.himalaya
…
   Compiling lettre v0.10.0-beta.3
   Compiling toml v0.5.8
   Compiling himalaya v0.2.6 (/build/source)
    Finished release [optimized] target(s) in 2m 02s
Executing cargoInstallPostBuildHook
Finished cargoInstallPostBuildHook
Finished cargoBuildHook
installing
Executing cargoInstallHook
/nix/store/11r443cfz2pnv2w8cp9jmmbxrabwbw4v-stdenv-linux/setup: line 84: /nix/store/1mn2v71dihhyh2gi16mjg822j0fnw030-himalaya-0.2.6-aarch64-unknown-linux-gnu/bin/himalaya: No such file or directory
builder for '/nix/store/36gr1fp7n41a3f31vanawc8i2vvf5xj5-himalaya-0.2.6-aarch64-unknown-linux-gnu.drv' failed with exit code 127
error: build of '/nix/store/36gr1fp7n41a3f31vanawc8i2vvf5xj5-himalaya-0.2.6-aarch64-unknown-linux-gnu.drv' failed

bt1cn@pcclu01:~/devel/nixpkgs$ file /nix/store/1mn2v71dihhyh2gi16mjg822j0fnw030-himalaya-0.2.6-aarch64-unknown-linux-gnu/bin/himalaya
/nix/store/1mn2v71dihhyh2gi16mjg822j0fnw030-himalaya-0.2.6-aarch64-unknown-linux-gnu/bin/himalaya: ELF 64-bit LSB shared object, ARM aarch64, version 1 (SYSV), dynamically linked, interpreter /nix/store/fjkh3sqz7z5n47na17zsd4wg9x9xypx4-glibc-2.32-40-aarch64-unknown-linux-gnu/lib/ld-linux-aarch64.so.1, for GNU/Linux 2.6.32, with debug_info, not stripped

bt1cn@pcclu01:~/devel/nixpkgs$ arch
x86_64
```

I think it would be better to add an option for this and enable it by default if buildPlatform == hostPlatform. Something like this:

```patch
-{ lib, stdenv, rustPlatform, fetchFromGitHub, openssl, pkg-config, installShellFiles }:
+{ lib, stdenv, rustPlatform, fetchFromGitHub, openssl, pkg-config, installShellFiles, enableCompletions ? stdenv.hostPlatform == stdenv.buildPlatform }:
```

```patch
-  postInstall = ''
+  postInstall = lib.optionalString enableCompletions ''
```

Reply:
Thanks for suggestion.